packages/cli/test/integration/project.api.test.ts

```diff
@@ -432,6 +433,32 @@ describe('POST /projects/', () => {
 
 		expect(await Container.get(ProjectRepository).count({ where: { type: 'team' } })).toBe(2);
 	});
+
+	test('should respect the quota when trying to create multiple projects in parallel (no race conditions)', async () => {
+		const globalConfig = Container.get(GlobalConfig);
+		// Preventing this relies on transactions and we can't use them with the
+		// sqlite legacy driver due to data loss risks.
+		if (dbType === 'sqlite' && globalConfig.database.sqlite.poolSize === 0) {
+			return expect(1).toBe(1);
+		}
```

Comment:
Should we extract this into a separate test and mock the DB type and pool size?
I think tests should avoid branching logic for clarity

Reply:
I'm not sure I understand what that would help with. 

This is only there to make the test not test anything on the legacy driver, because it's not fixed for the legacy driver. I could skip adding the test altogether, e.g. 

```ts
	// Preventing this relies on transactions and we can't use them with the
	// sqlite legacy driver due to data loss risks.
	if (dbType !== 'sqlite' || globalConfig.database.sqlite.poolSize !== 0) {
		test('should respect the quota when trying to create multiple projects in parallel (no race conditions)', async () => {
			// ...
		});
	}

```